src/pool/srv_pool.c

```diff
@@ -5427,32 +5431,57 @@ pool_svc_update_map_internal(struct pool_svc *svc, unsigned int opc,
 	if (rc != 0)
 		goto out_lock;
 
-	/*
-	 * If an empty target ID list is provided, convert from target
-	 * addresses.
-	 */
-	if (tgts->pti_number == 0) {
-		D_ASSERT(tgts->pti_ids == NULL);
-		D_ASSERT(tgt_addrs != NULL);
-		D_ASSERT(inval_tgt_addrs != NULL);
-		rc = pool_find_all_targets_by_addr(map, tgt_addrs, tgts,
-						   inval_tgt_addrs);
+	if (opc == POOL_EXTEND) {
```

Comment:
hmm, extend is a bit different as other pool map update ops, since extend needs to add domains/targets to the pool map,  then update the target status, while other operations only update target status.  So I tend to think this if () means to generate the new pool map and extract the target lists,   then call ds_pool_map_tgts_update() to update the status.

Though if we have better algorithm for extending, it might be clearer to do extend in ds_pool_map_tgts_update() indeed.  I will check what I can do in the following patches.